### PR TITLE
Tls

### PR DESCRIPTION
**Pull request by Clay-Ratliff-Redis**
Kyle added functionality to allow support for using TLS connections and a flag to allow self-signed certificates when connecting with gears-cli.

Default behavior when using tls connections is to require valid certs.  the self-signed flag must be explicitly passed to turn off cert validation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,6 +91,8 @@ Usage: gears-cli install-requirements [OPTIONS] [REQUIREMENTS]...
 Options:
   --host TEXT               Redis host to connect to
   --port INTEGER            Redis port to connect to
+  --ssl                     the connection will be made over ssl
+  --no-verify-certs         certificate validation will be skipped
```
Comment (MeirShpilraien) [inline README.md:95]: Why we do not give an option to give the cert file?

File: gears_cli/__main__.py [hunk 2]
```diff
@@ -45,10 +45,13 @@ def Green(data):
...
 
-def create_connection(host, port, password, decode_responses=True):
+def create_connection(host, port, ssl, no_verify_certs, password, decode_responses=True):
     global args
+    ssl_cert_reqs = 'required'
+    if no_verify_certs:
+        ssl_cert_reqs = None
     try:
-        r = redis.Redis(host, port, password=password, decode_responses=decode_responses)
+        r = redis.Redis(host, port, password=password, ssl=ssl, ssl_cert_reqs=ssl_cert_reqs, decode_responses=decode_responses)
```
Comment (MeirShpilraien) [inline gears_cli/__main__.py:54]: Isn't `ssl_cert_reqs` should be a path to cert file?

**Review by MeirShpilraien — COMMENTED** [review]
I do not see an option to give cert file, what am I missing?

**Comment by deji-alaran** [comment]
@MeirShpilraien This is client side only.  There is no need to give a cert file.  ssl_cert_reqs is actually a string that is either "required" to perform hostname validation or "none" not to.

**Comment by deji-alaran** [comment]
Is it possible to get this merged as I need the functionality?

**Comment by MeirShpilraien** [comment]
@deji-alaran I still think we should allow providing a certificate file to make the feature complete.

**Comment by MeirShpilraien** [comment]
Handled here: https://github.com/RedisGears/gears-cli/pull/32

**Pull request closed without merging**